packages/runtime/src/next/tests/server.makeswift-api-handler.test.ts

```diff
@@ -0,0 +1,207 @@
+import { createRequest, createResponse } from "node-mocks-http"
+import { randomUUID } from "crypto"
+import * as nextCache from 'next/cache'
+
+import { MakeswiftApiHandler } from "../api-handler"
+import { NextApiRequest, NextApiResponse } from 'next'
+import { ReactRuntime } from "../../react"
+
+const apiKey = 'fake-api-key'
+
+type MakeswiftApiHandlerArgs = Partial<Parameters<typeof MakeswiftApiHandler>[1]>
+
+function createHandler(args: Partial<MakeswiftApiHandlerArgs> = {}) {
+  const runtime = new ReactRuntime()
+  return MakeswiftApiHandler(apiKey, {
+    ...args,
+    runtime,
+    apiOrigin: 'https://api.fakeswift.com',
+    appOrigin: 'https://app.fakeswift.com',
+  })
+}
+
+jest.mock('next/cache', () => ({
+  revalidateTag: jest.fn(),
+}))
+
+afterEach(() => {
+  jest.restoreAllMocks()
+})
+
+describe('MakeswiftApiHandler', () => {
+  let consoleErrorSpy: jest.SpyInstance
+
+  beforeEach(() => {
+    consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => { });
+  });
+
+  describe('/api/makeswift/revalidate', () => {
+    function revalidateRequest(
+      query: Record<string, string | string[]> = {},
+    ) {
+      return createRequest<NextApiRequest>({
+        method: 'GET',
+        query: { makeswift: ['revalidate'], ...query },
```

Comment:
These should probably be set in the URL not directly in the query? Unless I'm missing how this utility function is intended to work?

What I mean is, the URL of the request should be `/api/makeswift/revalidate`. Passing it in the query makes some assumptions about Next.js routing internals (and honestly probably shouldn't work)

Reply:
I could not find a tool that can be used to simulate how Next.js does this thing. I will make a little helper to make it use URL instead of a structure.